src/modules/python/pbs_hooks/PBS_xeon_phi_provision.PY

```diff
@@ -0,0 +1,72 @@
+import pbs
+import sys
+import os
+import time
+
+sys.path.append('/opt/pbs/python/lib/python2.7')
+import subprocess
+import json as json
+
+# Section to be moved to config file once available for provision hook
+capmc_dir = '/opt/cray/capmc/default/bin'
+power_down_checks = 10
+power_down_sleep = 15
+power_up_checks = 20
+power_up_sleep = 60
+
+# Get the information about the node and aoe to provision
+e = pbs.event()
+vnode = e.vnode
+aoe = e.aoe
+
+numa_cfg, cache_percent = aoe.split('_')
+nid = vnode.split('_')[1]
+
+# Add capmc_dir to path
+os.environ['PATH'] = capmc_dir + ":" + os.environ['PATH']
+
+# capmc commands to configure the knl nodes
+cmd_set_numa_cfg = 'capmc set_numa_cfg --nids %s --mode %s' % (nid, numa_cfg)
+cmd_set_mcdram_cfg = 'capmc set_mcdram_cfg --nids %s --mode %s' % (
+    nid, cache_percent)
+cmd_node_reinit = 'capmc node_reinit --nids %s' % (nid)
+cmd_node_status = ['capmc', 'node_status', '--nids', '%s' % (nid)]
+
+pbs.logmsg(pbs.EVENT_DEBUG3, "vnode: %s" % vnode)
+pbs.logmsg(pbs.EVENT_DEBUG3, "aoe: %s" % aoe)
+pbs.logmsg(pbs.EVENT_DEBUG3, "numa cmd: %s" % cmd_set_numa_cfg)
+pbs.logmsg(pbs.EVENT_DEBUG3, "cache cmd: %s" % cmd_set_mcdram_cfg)
+pbs.logmsg(pbs.EVENT_DEBUG3, "power reinit cmd: %s" % cmd_node_reinit)
+
+appret1 = os.system(cmd_set_numa_cfg)
+pbs.logmsg(pbs.EVENT_DEBUG3, "appret1: %s" % str(appret1))
```

Comment:
please change it from appret1,2,3 to what it is actually executing

Reply:
Done